--- <unnamed>
+++ <unnamed>
@@ -3,7 +3,7 @@
 /**
  * Test the persistent object cache using core's cache tests
  */
-class CacheTest extends WP_UnitTestCase {
+abstract class CacheTestCase extends WPRedisTestCase {

 	private $cache;

@@ -28,12 +28,18 @@
 		'retry_interval' => 100,
 	);

-	public function setUp() {
+	public function setUp(): void {
 		parent::setUp();
 		$GLOBALS['redis_server'] = array(
 			'host' => '127.0.0.1',
 			'port' => 6379,
 		);
+
+		$GLOBALS['redis_server']['host'] = 'redis';
+		self::$client_parameters['host'] = 'redis';
+		$GLOBALS['wp_object_cache'] = new TaroskyObjectCache;
+		$GLOBALS['wp_object_cache']->flush();
+
 		// create two cache objects with a shared cache dir
 		// this simulates a typical cache situation, two separate requests interacting
 		$this->cache               =& $this->init_cache();
@@ -1361,7 +1367,7 @@
 		$this->cache->perform_client_connection( $redis, $settings, $keys_methods );
 	}

-	public function tearDown() {
+	public function tearDown(): void {
 		parent::tearDown();
 		$this->flush_cache();
 	}
@@ -1369,7 +1375,7 @@
 	/**
 	 * Remove the object-cache.php from the place we've dropped it
 	 */
-	static function tearDownAfterClass() {
+	static function tearDownAfterClass(): void {
 		// @codingStandardsIgnoreStart
 		unlink( ABSPATH . 'wp-content/object-cache.php' );
 		// @codingStandardsIgnoreEnd
